x-pack/plugins/lens/public/visualizations/xy/load_annotation_library_flyout.tsx

```diff
@@ -0,0 +1,106 @@
+
+export function LoadAnnotationLibraryFlyout({
+  eventAnnotationService,
+  isLoadLibraryVisible,
+  setLoadLibraryFlyoutVisible,
+  addLayer,
+}: {
+  isLoadLibraryVisible: boolean;
+  setLoadLibraryFlyoutVisible: (visible: boolean) => void;
+  eventAnnotationService: EventAnnotationServiceType;
+  addLayer: (argument: ExtraAppendLayerArg) => void;
+}) {
+  const containerPanelRef = useRef<HTMLDivElement | null>(null);
+  const otherElementsHeight = 250;
+  const singleEntryHeight = 40;
+  const numberOfElements = Math.floor(
+    ((containerPanelRef?.current?.clientHeight || 800) - otherElementsHeight) / singleEntryHeight
+  );
+
+  const [selectedItem, setSelectedItem] = React.useState<{
+    id: string;
+    type: string;
+    fullName: string;
+    savedObject: SavedObjectCommon<unknown>;
+  } | null>(null);
+
+  // needed to clean the state when clicking not on the item on the list
+  const hasBeenClicked = useRef(false);
+
+  React.useEffect(() => {
+    hasBeenClicked.current = false;
+  }, [selectedItem]);
+
+  const {
+    renderEventAnnotationGroupSavedObjectFinder: EventAnnotationGroupSavedObjectFinder,
+    loadAnnotationGroup,
+  } = eventAnnotationService || {};
+
+  return (
+    <FlyoutContainer
+      panelContainerRef={(el) => (containerPanelRef.current = el)}
+      customFooter={
+        <EuiFlyoutFooter className="lnsDimensionContainer__footer">
+          <EuiFlexGroup
+            responsive={false}
+            gutterSize="s"
+            alignItems="center"
+            justifyContent="spaceBetween"
+          >
+            <EuiFlexItem grow={false}>
+              <EuiButtonEmpty
+                flush="left"
+                size="s"
+                iconType="cross"
+                onClick={() => {
+                  setLoadLibraryFlyoutVisible(false);
+                  setSelectedItem(null);
+                }}
+                data-test-subj="lns-indexPattern-loadLibraryCancel"
+              >
+                {i18n.translate('xpack.lens.loadAnnotationsLibrary.cancel', {
+                  defaultMessage: 'Cancel',
+                })}
+              </EuiButtonEmpty>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFlyoutFooter>
+      }
+      isOpen={Boolean(isLoadLibraryVisible)}
+      groupLabel={i18n.translate('xpack.lens.editorFrame.loadFromLibrary', {
+        defaultMessage: 'Select annotations from library',
+      })}
+      handleClose={() => {
+        setLoadLibraryFlyoutVisible(false);
+        setSelectedItem(null);
+        return true;
+      }}
+    >
+      <div className="lnsIndexPatternDimensionEditor--padded">
+        <EventAnnotationGroupSavedObjectFinder
+          onChoose={({ id }) => {
+            loadAnnotationGroup(id).then((loadedGroup) => {
+              addLayer({ ...loadedGroup, annotationGroupId: id });
+              setLoadLibraryFlyoutVisible(false);
+            });
+          }}
+          fixedPageSize={numberOfElements}
```

Comment:
What exactly does this setting?

Reply:
Great question. It looks like all this was an effort on Marta's part to match the page size to the height of the flyout so that the user wouldn't have to scroll to see the rest of the content.
<img width="560" alt="Screenshot 2023-05-18 at 3 46 42 PM" src="https://github.com/elastic/kibana/assets/315764/72b1bc0e-9c82-4ac8-ac1b-e1263064ec01">


I like the idea, but it doesn't currently seem to work since the height of the container is determined only after the flyout is rendered of course. I'm not sure if this idea is worth pursuing further cc @MichaelMarcialis

Reply:
++ on removing this for now